### ISSUE TITLE
Use TypeChecker to infer dimensions of tmp fields

### ISSUE THREAD

**Issue by Stagno** [MeteoSwiss-APN/dawn#636]
Creating a temporary field on unstructured grids is currently unsupported (https://github.com/MeteoSwiss-APN/dawn/pull/602). By using the type checker we should be able to figure out the dimensions of whatever we want to assign to a temporary field, thus determining the dimensions of the temporary field, needed for construction. Passes involved are: `PassInilining`, `PassSetNonTempCaches`, `TemporaryHandling`.

**Comment by ofuhrer** [comment]
Minor comment: if this checker is used to infer dimensions, "TypeChecker" probably is no longer the right name.

**Comment by Stagno** [comment]
> Minor comment: if this checker is used to infer dimensions, "TypeChecker" probably is no longer the right name.

Agree. The implementation of the type checker can be decoupled into a routine that infers the type (dimensions actually) of an expression and into another one that does the checks. The former can be reused for this. Should this change the DoD of this task?

**Comment by Stagno** [comment]
After further discussion we decided it is better to postpone this task as this information would be available from the stage location type. But first we should fix the stage's location type #592
